libr/util/assert.c

```diff
@@ -0,0 +1,25 @@
+#include <r_util.h>
+
+#define S_SIZE 200
+
+R_API void r_log_warn(const char *file, int line, const char *func, const char *warnexpr) {
+	char s[S_SIZE];
+
+	s[0] = '\0';
+	if (warnexpr) {
+		snprintf (s, S_SIZE, "(%s:%d):%s%s runtime check failed: (%s)\n",
+			  file, line, func, func[0] ? ":" : "", warnexpr);
+	} else {
+		snprintf (s, S_SIZE, "(%s:%d):%s%s code should not be reached\n",
+			  file, line, func, func[0] ? ":" : "");
+	}
+	eprintf ("WARNING: %s", s);
+}
+
+R_API void r_log_return_warn(const char *func, const char *expr) {
+	eprintf ("WARNING: %s: assertion '%s' failed\n", func, expr);
```

Comment:
this can result on several badly placed warning messages because a precondition failing doesnt means an error

Reply:
A precondition failings should definitely be a warning! The purpose of these function is to detect programming errors as explained in https://developer.gnome.org/glib/stable/glib-Warnings-and-Assertions.html#g-return-if-fail . I should probably add that description somewhere in the comments.